### PR TITLE
#240/post method to add chatroom member

### PR DESCRIPTION
**Pull request by Damrod**
as detailed in #240

### CODE REVIEW

File: transcendence-app/src/chat/chat.controller.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class ChatController {
     return chat;
   }
 
-  @Post(':chatId')
+  @Post(':chatId/members')
```
Comment (vdedios) [inline transcendence-app/src/chat/chat.controller.ts:58]: We may want to have another endpoint, `/<chatId>/members/<userId>` so we can add other users apart from current? (I am thinking in an adming adding users to a chatroom)

File: transcendence-app/src/chat/chat.controller.ts [hunk 2]
```diff
@@ -49,6 +55,25 @@ export class ChatController {
...
   }
 
+  @Post(':chatId/members')
+  @ApiCreatedResponse({ description: 'Add a member to a chatroom', type: Chat })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiUnprocessableEntityResponse({ description: 'Unprocessable entity' })
+  async createChatRoomMember(
+    @GetUser() user: User,
+    @Param('chatId', ParseUUIDPipe) chatId: string,
+  ) {
```
Comment (gbudau) [inline transcendence-app/src/chat/chat.controller.ts:65]: praise: I like how you implemented this endpoint and structured the code.

**Review by gbudau — APPROVED** [review]
Thank you for this feature and nice implementation! LGTM :rocket:

File: transcendence-app/src/chat/chat.controller.ts [hunk 3]
```diff
@@ -49,6 +55,25 @@ export class ChatController {
     return chat;
   }
 
+  @Post(':chatId/members')
+  @ApiCreatedResponse({ description: 'Add a member to a chatroom', type: Chat })
```
[the remark this answers is not in the document]
Comment (Damrod) [reply transcendence-app/src/chat/chat.controller.ts:59]: I have done this locally, but after the changes, when attempting to regenerate the generated files, I get this error:

```
Errors: 
        -attribute components.schemas.ChatMember.default is not of type `object`

        at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:604)
        at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:631)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:457)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
```
Do you have any idea of why is it so?

[the remark this answers is not in the document]
Comment (gbudau) [reply transcendence-app/src/chat/chat.controller.ts]: IIRC, the generator requires types for all the members of the returned object, even if TypeScript can infer them.

<details>

```diff
diff --git a/transcendence-app/src/chat/chatmember.domain.ts b/transcendence-app/src/chat/chatmember.domain.ts
index c08ff61..60d838b 100644
--- a/transcendence-app/src/chat/chatmember.domain.ts
+++ b/transcendence-app/src/chat/chatmember.domain.ts
@@ -1,12 +1,13 @@
+/* eslint-disable @typescript-eslint/no-inferrable-types */
 import { ChatMemberEntity } from './infrastructure/db/chatmember.entity';
 
 export class ChatMember {
   chatId!: string;
   userId!: string;
   joinedAt!: Date;
-  admin = false;
-  muted = false;
-  banned = false;
+  admin: boolean = false;
+  muted: boolean = false;
+  banned: boolean = false;
   constructor(entity: ChatMemberEntity) {
     Object.assign(this, entity);
   }

```

</details>

[the remark this answers is not in the document]
Comment (Damrod) [reply transcendence-app/src/chat/chat.controller.ts]: it should work now

File: transcendence-app/src/chat/chat.controller.ts (hunk 2 above)
Comment (Damrod) [reply transcendence-app/src/chat/chat.controller.ts]: Thanks!

**Pull request merged**